### PR TITLE
Add resource usage to HttpRequest, add offline query param

### PR DESCRIPTION
**Pull request by tobrun**
Closes https://github.com/mapbox/mapbox-gl-native/issues/14819

Add resource usage to HttpRequest, add offline query parameter to HttpRequest.
This PR remains in draft until we have a PR open for iOS

cc @julianrex

### CODE REVIEW

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/http/HttpRequestUrl.java [hunk 1]
```diff
@@ -17,14 +17,19 @@ private HttpRequestUrl() {
...
     if (isValidMapboxEndpoint(host)) {
       if (querySize == 0) {
         resourceUrl = resourceUrl + "?";
       } else {
         resourceUrl = resourceUrl + "&";
       }
-      resourceUrl = resourceUrl + "sku=" + Mapbox.getSkuToken();
+
+      if (offline) {
+        resourceUrl = resourceUrl + "offline=true";
```
Comment (LukasPaczos) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/http/HttpRequestUrl.java]: I think we still want to send the sku token with offline requests for now, but I'm going to defer to @zugaldia on that.

Comment (tobrun) [reply platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/http/HttpRequestUrl.java]: This wasn't completely clear to me when reading the requirements. Any idea @zugaldia?

Comment (zugaldia) [reply platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/http/HttpRequestUrl.java]: Correct. We still want to attach the SKU token.

Comment (tobrun) [reply platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/http/HttpRequestUrl.java]: Code and tests have been updated to reflect above

File: platform/android/src/http_file_source.cpp [hunk 2]
```diff
@@ -70,6 +70,10 @@ HTTPRequest::HTTPRequest(jni::JNIEnv& env, const Resource& resource_, FileSource
       callback(callback_) {
     std::string etagStr;
     std::string modifiedStr;
+    bool offline = false;
+    if (resource_.usage == Resource::Usage::Offline) {
+        offline = true;
+    }
```
Comment (tobrun) [inline platform/android/src/http_file_source.cpp]: should be simplified to 1 line

**Comment by tobrun** [comment]
iOS landed in https://github.com/mapbox/mapbox-gl-native/pull/14857, this PR is ready for review.

**Pull request merged**